go/internal/cbor/deterministic_test.go

```diff
@@ -188,6 +188,48 @@ func TestByteStringIsNotDeterministic(t *testing.T) {
 	})
 }
 
+func TestArrayIsDeterministic(t *testing.T) {
+	arr := multiappend([]byte{0b10000101}, uint23, uint24, uint45, uint4294967295, uint18446744073709551615)
+	cborSequenceArray := multiappend(arr, arr)
+
+	for _, testCase := range [][]byte{arr, cborSequenceArray} {
+		err := Deterministic(testCase)
+		if err != nil {
+			t.Error("Deterministically encoded CBOR array should not return false for deterministicy.")
+		}
+	}
+}
+
+func TestEmptyArrayIsDeterministic(t *testing.T) {
+	err := Deterministic([]byte{0b10000000})
+	if err != nil {
+		t.Error("Empty array should not return false for deterministicy.")
+	}
+}
+
+func TestLongArrayIsDeterministic(t *testing.T) {
+	const numOfItems = 24
+	longArr := make([]byte, 2 /*=num of startBytes*/ +len(uint45)*numOfItems)
+	longArr[0] = 0b10011000 // ARR (4) + ONE_BYTE (24)
+	longArr[1] = byte(numOfItems)
+
+	// Fill in the byte array with 24 times uint45 bytes.
+	for i := 2; i < len(longArr); i += len(uint45) {
+		for j := 0; j < len(uint45); j++ {
+			longArr[i+j] = uint45[j]
+		}
+	}
+
+	cborSequenceArray := multiappend(longArr, longArr)
+
+	for _, testCase := range [][]byte{longArr, cborSequenceArray} {
+		err := Deterministic(testCase)
+		if err != nil {
+			t.Error("Deterministically encoded CBOR array should not return false for deterministicy.")
+		}
+	}
+}
+
```

Comment:
Can we have some negative test cases? (e.g. an array containing a non-deterministically-encoded CBOR item).

Reply:
I added one with non-deterministic item and another with too little items :)